packages/swing-store/src/snapStore.js

```diff
@@ -154,72 +155,30 @@ export function makeSnapStore(
           snapReader.destroy();
         });
 
-        // Create a file for the compressed snapshot in-place
-        // to be atomically renamed once we know its name
-        // from the hash of raw snapshot contents.
-        const {
-          path: tmpGzPath,
-          fd: tmpGzFd,
-          cleanup: tmpGzCleanup,
-        } = await ptmpFile({ tmpdir: root });
-        cleanup.push(tmpGzCleanup);
-        const gzWriter = createWriteStream(noPath, {
-          fd: tmpGzFd,
-          autoClose: false,
-        });
-        cleanup.push(() => gzWriter.close());
-
-        await Promise.all([fsStreamReady(snapReader), fsStreamReady(gzWriter)]);
+        await fsStreamReady(snapReader);
 
         const hashStream = createHash('sha256');
         const gzip = createGzip();
+        let compressedByteCount = 0;
 
         const { result: hash, duration: compressSeconds } =
           await measureSeconds(async () => {
             snapReader.pipe(hashStream);
-            snapReader.pipe(gzip).pipe(gzWriter);
-
-            await Promise.all([finished(snapReader), finished(gzWriter)]);
+            const snapshot = await buffer(snapReader.pipe(gzip));
+            await finished(snapReader);
```

Comment:
I'm not sure how the streams work, but.. does seeing `snapReader` finish guarantee that `hashStream` has *also* finished? It seems to me the inbound stream will push work to a couple of different places, and there will be one Promise that fires when that inbound stream has sent EOF, but there's some different Promise that fires to feed the last bit of data to the hashing pipe, and we don't know for sure that it's received that data yet.

Reply:
if `snapReader` has finished that means it has piped everything into `hashStream`.

Also this PR does not change the previous behavior in any way regarding the hash stream.

Reply:
But could `hashStream` still be waiting on that pipe for the last bit of input? I'm thinking about which turns of the microtask queue have been executed by this point, and knowing that `snapReader` has enqueued a turn to `hashStream` doesn't necessarily mean that `hashStream` has executed that turn and updated the hash yet. Just worried because this would be a really really nasty bug (silently omitting the last chunk of data from the hash, mostly in consensus, maybe allowing an attacker to replace the last chunk of a heap snapshot when distributing a state-sync snapshot).

Reply:
Node streams are not based on promises. A piped into stream will get receive all data synchronously from upstream.

Reply:
ah, gotcha, that's what I was missing.. thanks!